File: 1844-maximum-number-of-balls-in-a-box/maximum-number-of-balls-in-a-box.js
/**
 * @param {number} lowLimit
 * @param {number} highLimit
 * @return {number}
 */

var digSum = (num) => {
    let ans = 0;
    while(num > 0){
        ans += num%10;
        num = Math.floor(num/10);
    }
    return ans;
}

var countBalls = function(lowLimit, highLimit) {
    let ans = 0;
    let map = {};
    for(let i=lowLimit; i<=highLimit; ++i){
        let cur = digSum(i);
        map[cur] = (map[cur] || 0) + 1;
        ans = Math.max(ans, map[cur]);
    }
    return ans;
};

File: 1844-maximum-number-of-balls-in-a-box/maximum-number-of-balls-in-a-box.java
class Solution {
    static int digSum(int x){
        int ans = 0;
        while(x > 0){
            ans += x%10;
            x /= 10;
        }
        return ans;
    }

    public int countBalls(int lowLimit, int highLimit) {
        int ans = 0;
        HashMap<Integer, Integer> map = new HashMap<>();

        for(int i=lowLimit; i<=highLimit; ++i){
            int cur = digSum(i);
            map.put(cur, map.getOrDefault(cur, 0) + 1);
            ans = Math.max(ans, map.getOrDefault(cur, 0));
        }

        return ans;
    }
}